plugins/reporters/spdx/src/funTest/kotlin/SpdxDocumentReporterFunTest.kt

```diff
@@ -119,6 +120,19 @@ class SpdxDocumentReporterFunTest : WordSpec({
                 custom = fromYaml<SpdxDocument>(yamlSpdxDocument).getCustomReplacements()
             )
         }
+
+        "create the expected document for the ORT result of a Go project" {
+            val ortResultFile = getAssetFile("disclosure-cli-analyzer-and-scanner-result.yml")
+            val ortResultForGoProject = ortResultFile.readValue<OrtResult>()
+            val expectedResultFile = getAssetFile("disclosure-cli-expected-output.spdx.yml")
+
+            val yamlSpdxDocument = generateReport(ortResultForGoProject, FileFormat.YAML)
+
+            yamlSpdxDocument should matchExpectedResult(
```

Comment:
Can we reduce the assertion to 'Map<String, String>` `(SPDX-id, license-declared)`, to not assert that much?

(And also rename the test case) ?

Reply:
Of course we could, but I do not see a reason to do so. As I wrote elsewhere, I prefer to align to similar tests in the same file. Also, a positive side effect of making the test more complete is that issues for which the test was not originally designed might get discovered as well. A concrete example is the purl encoding for Go projects, which recently was changed and got catched by this test.

Reply:
> Of course we could, but I do not see a reason to do so.

Wouldn't it be easier to maintain?

Reply:
> Wouldn't it be easier to maintain?

In what regard? It's very little code, so code-wise probably not. Or are you afraid that expected results need to often be updated (for no good reason)? I believe that risk to be rather low for this test, as if something needs updating here, it's probably a change we *want* to be aware of. But if I'm wrong and this becomes one of the tests that needs updates all of the time for no good reason, we could still reduce the test scope.

Reply:
>  it's probably a change we want to be aware of

I intended to speak about redundant testing, so if the test is about showing to get the declared license right, then it's IMO asserting too much.

Reply:
> it's IMO asserting too much.

I totally agree that the test is asserting more than what's strictly required if you'd only want to assert the `licenseDeclared` field. But as said, I've deliberately chosen this to be a more generic YAML output test so that also other aspects are tested, esp. as the existing tests were only using synthetic ORT result data instead of real-life data.